packages/neuron-ui/src/components/HDWalletSign/index.tsx

```diff
@@ -56,21 +56,12 @@ const HDWalletSign = ({ tx }: { tx: State.DetailedTransaction }) => {
         } else {
           try {
             if (cell.lock.codeHash === systemCodeHash && cell.lock.hashType === 'type') {
-              address = ckbCore.utils.bech32Address(cell.lock.args, {
-                prefix: addressPrefix,
-                type: ckbCore.utils.AddressType.HashIdx,
-                codeHashOrCodeHashIndex: '0x00',
-              })
+              address = generateAddress(cell.lock, { config })
```

Comment:
The address is not derived from the lock, it's actually derived from
```yaml
code_hash: SECP256k1_blake160 # https://github.com/ckb-js/ckb-sdk-js/blob/develop/packages/ckb-sdk-utils/src/address/index.ts#L85
hash_type: type
args: lock.args
```

This usage comes from a deprecated format mentioned at https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0021-ckb-address-format/0021-ckb-address-format.md#deprecated-short-payload-format

Reply:
Let's double check before commit, here's the new code.
```ts
if (cell.lock.codeHash === systemCodeHash && cell.lock.hashType === 'type') {
  address = generateAddress(
    {
      args: cell.lock.args,
      codeHash: config.SCRIPTS.SECP256K1_BLAKE160.CODE_HASH,
      hashType: config.SCRIPTS.SECP256K1_BLAKE160.HASH_TYPE,
    },
    { config }
  )
}
```
FYI, `generateAddress` generates deprecated addresses which matches the original code bahavior.

Reply:
My bad, seems that the condition of the `if-branch` has limited the `lock.codeHash` and `lock.hashType` to `config.SCRIPTS.SECP256K1_BLAKE160.CODE_HASH` and `config.SCRIPTS.SECP256K1_BLAKE160.HASH_TYPE` so no change is required